### PR TITLE
Added all features needed for programmer access

### PR DESCRIPTION
**Pull request by zadelobo**
Functions to read UDP packets, parse game state packages, parse controller state packages, and connect to Wi-Fi

### CODE REVIEW

File: robot-code/micro-python/client_udp_listen.py [hunk 1]
```diff
@@ -1,11 +1,78 @@
 import socket
+import network
+import time
 
+ROBOT_NAME = "St. Francis"
```
Comment (zeroDecoder) [inline robot-code/micro-python/client_udp_listen.py]: These attributes should probably go in a class and be part of the constructor

File: robot-code/micro-python/client_udp_listen.py [hunk 2]
```diff
@@ -1,11 +1,78 @@
...
+
+
+### Parses the UDP data to apply the game state to the robot if applicable
+### Returns a message with the game state for user info, no action required with this information
+def parse_game_state(data):
+	hex_list = [hex(byte) for byte in data]
+	game_state = data.pop(0)
+
+	#find out if the game state command is applicable to this robot
+	for elem in [hex_list[i:i+16] for i in range(0, len(hex_list), 16)]:
```
Comment (zeroDecoder) [inline robot-code/micro-python/client_udp_listen.py]: double for loop nesting in one line is bad for readability and makes this comment hard. You included the first byte of the message in the range so all the names will be off by one.

File: robot-code/micro-python/client_udp_listen.py [hunk 3]
```diff
@@ -1,11 +1,78 @@
...
+		"trigger_left": controller_data[4],
+		"trigger_right": controller_data[5],
+		"button_data": str(controller_data[6])+str(controller_data[7])
+	}
+
+	return controller_state
+
+### Sets up the ESP32 dev board to connect to the Wi-Fi network
+### Does not return and is not user-facing
+def connect_to_wifi():
```
Comment (zeroDecoder) [inline robot-code/micro-python/client_udp_listen.py]: This should get all this in a class

**Review by zeroDecoder — CHANGES_REQUESTED** [review]

**Comment by zadelobo** [comment]
@zeroDecoder pushed the changes requested to the python-client branch. Requesting another code review.

**Review by zeroDecoder — APPROVED** [review]
LGTM

**Pull request merged**